### ISSUE TITLE
We should rename the Upload plugin to Media Library in the Permissions view both for the Admin Roles and Users & Permissions Roles

### ISSUE THREAD

**Issue by yanniskadiri** [strapi/strapi#20842]
![Image](https://github.com/user-attachments/assets/762803a8-9345-4796-a511-16cef185aa07)

**Comment by WardenCommander** [comment]
we should move upload plugin outside core, and make it its own plugin to allow option to customize it and expand

**Comment by jhoward1994** [comment]
I was looking into this quickly and I think it is trickier than it seems.

As I understand it, the plugin name used here comes from this part of the upload plugin package json:
https://github.com/strapi/strapi/blob/77e66829c8c9c19a6fd9c80944bb2395c7d7f714/packages/core/upload/package.json#L117-L118

Changing this to e.g. 'media-library' would incur a lot of changes across Strapi wherever the upload plugin is referenced. & could also be a breaking change for plugin developers referencing the upload plugin?

Alternatively we could make specific edge cases in the code where we want to change the display name and reference this issue.

Unless theres something I'm missing. wdyt?

**Comment by yanniskadiri** [comment]
No, I think you are right @jhoward1994. Since it's a low-severity issue, we don't have to address it for now, especially if it introduces a breaking change. I agree we could make specific edge cases just to change the display name.

**Comment by Adzouz** [comment]
Hi there!
Jumping on the subject: don't we just want to change the localised string here? We don't need to completely rename the upload core folder for now and at least it will better on front side.
Let me know what you think!

**Comment by yanniskadiri** [comment]
I agree with this approach - if it’s not too complex, updating the localized string to “Media Library” in the admin panel sounds like a good first step. Ideally, we’d have a consistent name across both the admin and the codebase, but starting with the UI would already improve clarity for users. No need to go further if the full rename introduces too much complexity for now.

**Comment by Adzouz** [comment]
Hi @yanniskadiri, I just merged the fix and it will be part of the next release of Wednesday (5.1.6.0). Let me know if anything 😊
Have a great day 🚀
